frontend/src/app/navigation-bar/navigation-bar.component.ts

```diff
@@ -28,6 +29,19 @@ export class NavigationBarComponent implements OnInit {
         });
     }
 
+    @HostListener('window:scroll', [])
+    hideMenuOnScroll() {
+        if (this.menu && this.menu.overlayVisible) {
+            this.menu.hide();
+        }
+
+        // This makes hiding instant
+        const overlay = document.querySelector('.p-menu-overlay');
+        if (overlay) {
+            (overlay as HTMLElement).style.display = 'none';
+        }
```

Comment:
@limcaaarl Is this bit needed? I think it's generally not recommended to manipulate the DOM directly.